Include a Dockerfile and instructions.

Beanow opened the pull request:
Along the same lines of https://github.com/sourcecred/sourcecred/pull/1288

It's unlikely this repository will remain as a separate docker image for long as it would make sense to integrate it as a plugin that runs during the envisioned `sourcecred update`. But @vsoch mentioned her goal with using the docker containers is to generate the widgets. So this would allow that.

I've set the entrypoint to `/code/bin/contributor-wall-svg.js` as this is currently the only type of widget being supported. Probably not a very future-proof CLI, but hey, what else would you expect from `v0.0.1-alpha`? :man_shrugging:

Beanow commented:
As mentioned in sourcecred/sourcecred#1288
We may need to rename the images to `sourcecred/sourcecred` and `soucrecred/widgets` if we're pushing these to Docker Hub.

teamdandelion commented on .gitignore:
```diff
@@ -1 +1,2 @@
 node_modules
+.token
```
I'd prefer to use env rather than `.token` files, for the reasons discussed by @wchargin here:  https://github.com/sourcecred/sourcecred/pull/1288#discussion_r314375371

Beanow replied:
Missed that as it was resolved already there. Removed those.

vsoch replied:
A comment from https://github.com/sourcecred/sourcecred/pull/1288 that I'll include here too - instead of doing an export it might be advisable to just one off define the variable for each command:

```bash
SOURCECRED_GITHUB_TOKEN=YOUR_GITHUB_TOKEN docker run --rm -ti -v sourcecred_data:/data -e SOURCECRED_GITHUB_TOKEN sourcecred load sourcecred/sourcecred
```
I'm okay with the export (I don't share a system with anyone), but this was suggested for security.

Still testing :)

vsoch replied:
okay one quick point of feedback - there isn't any instruction to build the container, so I get to the line where I need `sourcecred-widgets` and I'm like :thinking: huh? We should add this, before it is needed:

Next, build the "sourcecred-widgets" container from the Dockerfile in the repository:

```bash
$ docker build -t sourcecred-widgets .
```

vsoch replied:
Also we still need to have the GitHub token, so that should be added to the example command.

vsoch replied:
It's so beautiful!! 

![image](https://user-images.githubusercontent.com/814322/63170365-1ddf7a80-c007-11e9-9585-b7c96d2ae326.png)

Screenshot of an svg, sorry about that :)

@Beanow when we get this merged, I can do another automated build for it, and then guess what! We can make GitHub actions too! I'm so pumped.

vsoch commented:
Oh let me test! So excited about this one :)

Beanow commented:
Added the image build step, removed the ENV export from security comments on the other PR, and use the sourcecred/* names for images.

With that I think we should be good to go :]

wchargin commented on README.md:
```diff
@@ -12,26 +12,31 @@ For example to use in README's or as webpage static assets.
-export SOURCECRED_GITHUB_TOKEN=YOUR_GITHUB_TOKEN
-node PATH_TO_SOURCECRED/bin/sourcecred.js load sourcecred/sourcecred
+SOURCECRED_GITHUB_TOKEN=YOUR_GITHUB_TOKEN \
+	node PATH_TO_SOURCECRED/bin/sourcecred.js load sourcecred/sourcecred
```
nit: SourceCred avoids tabs (same comment a few places below).

Beanow replied:
Fixed

wchargin commented on README.md:
````diff
@@ -12,13 +12,31 @@ For example to use in README's or as webpage static assets.
+```sh
+# 1. Use the main sourcecred image to export scores to a file.
+SOURCECRED_GITHUB_TOKEN=YOUR_GITHUB_TOKEN \
+	docker run --rm -ti -v sourcecred_data:/data -e SOURCECRED_GITHUB_TOKEN sourcecred/sourcecred load sourcecred/sourcecred
````
nit: These lines are long enough to be wrapping on my screen, which
makes the content ambiguous and hard to read; please hard-wrap at 80.

Beanow replied:
Applied here and https://github.com/sourcecred/sourcecred/pull/1288

vsoch commented:
@Beanow when we get this merged I'll open another like https://github.com/sourcecred/sourcecred/pull/1320 to get an automated build and deploy.

Beanow commented:
Currently the readme example relies on `sourcecred/sourcecred` being present. So I think it would be nice to set things up on dockerhub to manually deploy v0.4.0. I've held off on a merge for that.

Beanow commented:
Previous comment is resolved now that there's a manual build for sourcecred/sourcecred.
See https://github.com/sourcecred/sourcecred/issues/1331 / https://hub.docker.com/r/sourcecred/sourcecred/tags

This can be merged as well.
